### PR TITLE
Handle some promises better in cms

### PR DESCRIPTION
**Pull request by anthonydresser**
@aaomidi helped identify some promise hanging happening in connection management service.

### CODE REVIEW

File: src/sql/platform/connection/common/connectionManagementService.ts [hunk 1]
```diff
@@ -1148,33 +1148,26 @@ export class ConnectionManagementService extends Disposable implements IConnecti
...
 						this._onDeleteConnectionProfile.fire();
-						Promise.resolve(true);
-					}).catch(err => {
-						// Reject promise if error occurred writing to settings
-						Promise.reject(err);
+						return true;
 					});
 
 				} else {
 					// If connection fails to disconnect, resolve promise with false
```
Comment (Charles-Gagnon) [inline src/sql/platform/connection/common/connectionManagementService.ts]: nit: I'd just delete this comment. It's mostly useless

File: src/sql/platform/connection/common/connectionManagementService.ts [hunk 2]
```diff
@@ -1196,20 +1189,19 @@ export class ConnectionManagementService extends Disposable implements IConnecti
...
-		Promise.all(disconnected).then(() => {
+		return Promise.all(disconnected).then(() => {
 			// Remove profiles and groups from config
-			this._connectionStore.deleteGroupFromConfiguration(group).then(() => {
+			return this._connectionStore.deleteGroupFromConfiguration(group).then(() => {
 				this._onDeleteConnectionProfile.fire();
-				Promise.resolve(true);
+				return true;
 			}).catch(err => {
 				// If saving to config fails, reject promise with false
```
Comment (Charles-Gagnon) [inline src/sql/platform/connection/common/connectionManagementService.ts]: Same here - pretty much useless comment so suggest removing

File: src/sql/platform/connection/common/connectionManagementService.ts [hunk 3]
```diff
@@ -521,7 +521,7 @@ export class ConnectionManagementService extends Disposable implements IConnecti
 		if (!this.focusDashboard(connectionProfile)) {
 			let dashboardInput: DashboardInput = this._instantiationService ? this._instantiationService.createInstance(DashboardInput, connectionProfile) : undefined;
 			return dashboardInput.initializedPromise.then(() => {
-				this._editorService.openEditor(dashboardInput, { pinned: true }, ACTIVE_GROUP);
+				return this._editorService.openEditor(dashboardInput, { pinned: true }, ACTIVE_GROUP);
```
Comment (aaomidi) [inline src/sql/platform/connection/common/connectionManagementService.ts]: The function is returning a Thenable, shouldn't it just be async?

File: src/sql/platform/connection/common/connectionManagementService.ts [hunk 4]
```diff
@@ -395,7 +395,7 @@ export class ConnectionManagementService extends Disposable implements IConnecti
 		if (!tokenFillSuccess) {
 			throw new Error(nls.localize('connection.noAzureAccount', "Failed to get Azure account token for connection"));
 		}
-		return this.createNewConnection(uri, connection).then(connectionResult => {
+		return this.createNewConnection(uri, connection).then(async connectionResult => {
```
Comment (aaomidi) [inline src/sql/platform/connection/common/connectionManagementService.ts]: Same thing here, can't we await the connectionResult?

File: src/sql/platform/connection/common/connectionManagementService.ts (hunk 3 above)
Comment (anthonydresser) [reply src/sql/platform/connection/common/connectionManagementService.ts]: Certainly everywhere we use .then we could use async, but thats a larger change than what I'm looking to do here.

**Pull request merged**